### PR TITLE
ENH: interpolate: Use scipy.spatial.distance to speed-up Rbf

### PR DESCRIPTION
**Pull request by jrsassen**
The pairwise distances implemented in scipy.spatial are faster than the ones implemented in Rbf. Therefore, using them leads to a significant speed-up when using Rbf with a large number of points.

Previously a part of PR #9215

### CODE REVIEW

File: scipy/interpolate/rbf.py [hunk 1]
```diff
@@ -188,22 +185,20 @@ def _init_function(self, r):
         return a0
 
     def __init__(self, *args, **kwargs):
-        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten()
-                           for a in args[:-1]])
-        self.N = self.xi.shape[-1]
+        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten() for a in args[:-1]]).T
+        self.N = self.xi.shape[0]
```
Comment (rgommers) [inline scipy/interpolate/rbf.py]: `xi` and `N` are public attributes, they shouldn't be changed here. Looks to me like you can just add `.T` inside the calls to the distance functions instead.

Comment (jrsassen) [reply scipy/interpolate/rbf.py]: Done.

**Comment by jrsassen** [comment]
By the way, this should be labeled `scipy.interpolate` instead of `scipy.spatial`, right?

File: scipy/interpolate/rbf.py [hunk 2]
```diff
@@ -188,20 +185,18 @@ def _init_function(self, r):
         return a0
 
     def __init__(self, *args, **kwargs):
-        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten()
-                           for a in args[:-1]])
+        self.xi = np.asarray([np.asarray(a, dtype=np.float_).flatten() for a in args[:-1]])
```
Comment (larsoner) [inline scipy/interpolate/rbf.py]: Revert? The only change I see is taking a PEP8-compliant line and making it non-PEP8 compliant.

**Review by larsoner — COMMENTED** [review]
Other than one minor gripe, LGTM

**Comment by jrsassen** [comment]
While at it, I fixed some other PEP8 violations. Hope it's ok I added this here.

**Review by larsoner — APPROVED** [review]
LGTM +1 for merge. I am fine with incrementally getting to PEP8 compliance by changing nearby lines.

Feel free to ping me at the end of the week for merge unless someone else comments.

**Comment by rgommers** [comment]
Verified existing test coverage has the 1-D, 2-D, 3-D cases that should exercise the now removed code with `np.newaxis`. Looks good to me too now, merging. Thanks @jrsassen

**Pull request merged**